src/h5web/visualizations/nexus/utils.ts

```diff
@@ -71,3 +71,42 @@ export function getNxAxes(
 
   return { labels: axesLabels, ids: axesIds };
 }
+
+export function getDefaultEntity(
+  entity: HDF5Group,
+  metadata: HDF5Metadata
+): HDF5Entity | undefined {
+  const defaultPath = getAttributeValue(entity, 'default');
+
+  if (typeof defaultPath !== 'string') {
+    return undefined;
+  }
+
+  const relativePath = defaultPath.startsWith('/')
+    ? defaultPath.slice(1)
+    : defaultPath;
+
+  const defaultEntity = relativePath.split('/').reduce<HDF5Entity | undefined>(
+    (previousEntity, currentComponent) => {
+      if (!previousEntity) {
+        return undefined;
+      }
+      if (!isGroup(previousEntity)) {
+        return previousEntity;
+      }
+      return getLinkedEntity(previousEntity, metadata, currentComponent);
+    },
+    defaultPath.startsWith('/') ? metadata.groups[metadata.root] : entity
+  );
+
+  if (!defaultEntity || !isGroup(defaultEntity)) {
+    return undefined;
+  }
+
+  const nextDefaultPath = getAttributeValue(defaultEntity, 'default');
+  if (typeof nextDefaultPath !== 'string') {
+    return defaultEntity;
+  }
+
+  return getDefaultEntity(defaultEntity, metadata);
```

Comment:
I don't quite get what this function does and how the recursion algorithm. I feel like it might make more sense if it were called `getNxDataGroup` and if its job was to return either an `NXdata` group or `undefined`. Here is what the algorithm would look like in my mind:

1. If input entity is `undefined` or no a group, return `undefined`.
1. If entity is `NXdata` group, return entity.
1. If entity has no `default` attribute, return `undefined`; else recurse with target default entity.

Reply:
Given @vasole comment, I can simplify this function.